dot/parachain/types/view.go

```diff
@@ -0,0 +1,94 @@
+package parachaintypes
+
+import (
+	"reflect"
+	"sort"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// A succinct representation of a peer's view. This consists of a bounded amount of chain heads
+// and the highest known finalized block number.
+//
+// Up to `N` (5?) chain heads.
+type View struct {
+	// a bounded amount of chain Heads
+	Heads []common.Hash
+	// the highest known finalized number
+	FinalizedNumber uint32
+}
+
+// Difference returns hashes present in v View but not in v2 View.
+func (v View) Difference(v2 View) []common.Hash {
+
+	vHeads := SortableHeads(v.Heads)
+	v2Heads := SortableHeads(v2.Heads)
+
+	sort.Sort(vHeads)
+	sort.Sort(v2Heads)
```

Comment:
what if we create a method `AddHead` that inserts hashes in lexicographic order and here we don't need to sort them. Also would be good to have tests for this method

Reply:
> what if we create a method AddHead that inserts hashes in lexicographic order

I think you are asking me to make sure that heads are always sorted using `AddHead`, right?

If so, that's not possible. Since, heads could come from other nodes as part of `ViewUpdate`.

Reply:
I can add a test for SortableHeads, butI feel like the functionality is too trivial (like implement the sorting interface)